music21/environment.py

```diff
@@ -714,60 +714,8 @@ def launch(self, fmt, filePath, options='', app=None):
 
         Create a png, svg, etc. converter (or just a graphics converter) and call launch on it
         '''
-        # see common.fileExtensions for format names
-        filePath = common.cleanpath(filePath, returnPathlib=True)
-
-        m21Format, unused_ext = common.findFormat(fmt)
-        environmentKey = self.formatToKey(m21Format)
-        if environmentKey is None:
-            environmentKey = self.formatToKey(fmt)
-        if m21Format == 'vexflow':
-            try:
-                import webbrowser
-                filePath = filePath.as_uri()
-                webbrowser.open(filePath)
-                return
-            except ImportError:
-                print(f'Cannot open webbrowser, sorry. Go to file://{filePath}')
-        if app is not None:
-            # substitute app provided via argument
-            fpApp = app
-        elif environmentKey is not None:
-            fpApp = self._ref[environmentKey]
-        else:
-            fpApp = None
-
-        platform = common.getPlatform()
-        if fpApp is None:
-            if platform == 'win':
-                # no need to specify application here:
-                # windows starts the program based on the file extension
-                cmd = f'start {filePath}'
-            elif platform == 'darwin':
-                cmd = f'open {options} {filePath}'
-            else:
-                if m21Format == 'braille':
-                    with open(filePath, 'r') as f:
-                        for line in f:
-                            print(line, end='')
-                        print('')
-                    return
-                else:
-                    raise EnvironmentException(
-                        'Cannot find a valid application path '
-                        + f'for format {m21Format}. '
-                        + 'Specify this in your Environment by calling '
-                        + f"environment.set({environmentKey!r}, '/path/to/application')")
-        elif platform == 'win':  # note extra set of quotes!
-            cmd = f'""{fpApp}" {options} "{filePath}""'
-        elif platform == 'darwin':
-            cmd = f'open -a"{fpApp}" {options} "{filePath}"'
-        elif platform == 'nix':
-            cmd = f'{fpApp} {options} "{filePath}"'
-        else:
-            raise Exception(f'Unknown platform {platform}.')
-
-        os.system(cmd)
+        from music21.converter.subConverters import SubConverter
+        return SubConverter().launch(filePath, fmt=fmt, options=options, app=app)
```

Comment:
these are very similar routines, but not identical.  The SubConverter launch assumes that filePath has already been cleaned and that fmt has already been regularized.  Probably best to make SubConverter take over the safety checks that Enviornment once did.  Has this been tested for the tricky cases of graph.launch(), etc. listed in the docs?  (Also need docs updated to "undeprecate")  THANKS!

Reply:
`SubConverter` subclasses take care of cleaning paths in `.write()`, so I don't think we should add any more checks to subclass `launch` methods. Obviously steering this is up to you, but here's why I think these changes are safe, and get closer to deprecating `Environment.launch` altogether someday:

Four places call `Environment.launch`:
  - `graph.write`
  - `lily.showPNG`
  - `lily.showSVG`
  - `scale.show`
  - technically also `vexflow.toMusic21j.TextExternal.testCuthbertLocal`, but I've never heard of that guy

***

`graph.show` passes a hard-coded `'png'` format.
It usually receives a temp `fp` from `stream.plot()`, but a power user might pass an unexpanded path to `graph.show()`, so yes, we could add a safety check, but I would do it in `graph.write` to be analogous to `SubConverter.writeDataStream`.

***
`lily.showPNG` passes a hard-coded `'png'` format.
The `fp` passed to `launch` is the fp returned by `lily.createPNG`, which is a `pathlib.Path`, so I don't think it needs to be cleaned.

***
`lily.showSVG()`: sim.
***
`scale.show` already runs `common.findFormat()` and will only `launch` if regularized fmt == 'scala'.
The `fp` it passes to `launch` is returned from `self.write('scala')`, so since it's already been used, it's too late to clean it.

Reply:
I guess I could just finish the job and fully deprecate this!